lib/elixir/lib/module.ex

```diff
@@ -1190,8 +1190,12 @@ defmodule Module do
     end
   end
 
-  defp preprocess_attribute(:on_load, atom) when is_atom(atom) do
-    {atom, 0}
+  defp preprocess_attribute(:on_load, value) do
```

Comment:
The `@on_load` attribute can be either `{:fun, 0}` or `:fun`. So I think we should probably leave this clause intact and add a clause that check when it is not in the tuple format above.